### ISSUE TITLE
Segmentation issue with line "model.optimize_for_inference()"

### ISSUE THREAD

**Issue by hangxu811** [roboflow/rf-detr#466]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

I fine-tune rf-detr seg preview for my project, and I found out that when I deployed the model and do the inference without this line "model.optimize_for_inference()", everything works fine like below

detections = model.predict(image, threshold=0.3) 
print(detections)

Detections(xyxy=array([[888.2473 , 746.7094 , 932.44073, 768.10944]], dtype=float32), mask=array([[[False, False, False, ..., False, False, False],
        [False, False, False, ..., False, False, False],
        [False, False, False, ..., False, False, False],
        ...,
        [False, False, False, ..., False, False, False],
        [False, False, False, ..., False, False, False],
        [False, False, False, ..., False, False, False]]],
      shape=(1, 1024, 1280)), confidence=array([0.75640106], dtype=float32), class_id=array([1]), tracker_id=None, data={}, metadata={})

but when I implement this line of code "model.optimize_for_inference()", the model stops generating mask. 

Detections(xyxy=array([[888.24963, 746.70496, 932.4463 , 768.10974]], dtype=float32), mask=None, confidence=array([0.75613964], dtype=float32), class_id=array([1]), tracker_id=None, data={}, metadata={})

For people who want to train **segmentation**, please **do not** use this line of code as they use in their repo predict example. 

### Environment

RF-DETR 1.3

### Minimal Reproducible Example

custom_weights_path = "/root/runs/rfdetr_seg/checkpoint_best_regular.pth" 

try:
    model = RFDETRSegPreview(pretrain_weights=custom_weights_path)
except Exception as e:
    model = RFDETRSegPreview()

detections = model.predict(image, threshold=0.3) 

print(detections)

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by hangxu811** [comment]
> [@J4BEZ](https://github.com/J4BEZ) i believe it is already working in the raw code, just not in the release version

Thank you so much!

**Comment by isaacrob-roboflow** [comment]
Can I make this as completed @hangxu811 ?

**Comment by J4BEZ** [comment]
@isaacrob-roboflow 
Thank you! I'm delighted to have the opportunity to contribute to rf-detr.
I’ll open a PR soon!

**Comment by J4BEZ** [comment]
@isaacrob-roboflow 
Hello!

While I was working on preparing the PR, I noticed that another pull request addressing this exact issue has already been submitted.
- #429 
- #430 

In light of the solution already submitted by another contributor, I will refrain from submitting my PR to respect their effort and prevent redundant review cycles.

Thank you again for your hard work on this project, and I hope you have a peaceful day and stay healthy.

Best Regards, J4BEZ😊
